tests/instruments/agilent/test_agilent34450A.py

```diff
@@ -0,0 +1,488 @@
+from pyvisa.errors import VisaIOError
+
+pytest.skip('Only work with connected hardware', allow_module_level=True)
+
+
+class TestAgilent34450A:
+    """
+    Unit tests for Agilent34450A class.
+
+    An Agilent34450A device should be connected to the computer.
+    """
+
+    ###############################################################
+    # Agilent34450A device goes here:
+    resource = "USB0::10893::45848::MY56511723::0::INSTR" # EQ0017
+    ###############################################################
+
+    ############
+    # FIXTURES #
+    ############
+    @pytest.fixture
+    def make_reseted_dmm(self):
+        dmm = Agilent34450A(self.resource)
+        dmm.reset()
+        return dmm
+
+    #########################
+    # PARAMETRIZATION CASES #
+    #########################
+    BOOLEANS = [False, True]
+    RESOLUTIONS = [[3.00E-5, 3.00E-5], [2.00E-5, 2.00E-5], [1.50E-6, 1.50E-6],
+                   ["MIN", 1.50E-6], ["MAX", 3.00E-5], ["DEF", 1.50E-6]]
+    MODES = ["current", "ac current", "voltage", "ac voltage", "resistance",
+              "4w resistance", "current frequency", "voltage frequency",
+              "continuity", "diode", "temperature", "capacitance"]
+    CURRENT_RANGES = [[100E-6, 100E-6], [1E-3, 1E-3], [10E-3, 10E-3], [100E-3, 100E-3],
+                      [1, 1], ["MIN", 100E-6], ["MAX", 10], ["DEF", 100E-3]]
+    CURRENT_AC_RANGES = [[10E-3, 10E-3], [100E-3, 100E-3], [1, 1], ["MIN", 10E-3],
+                         ["MAX", 10], ["DEF", 100E-3]]
+    VOLTAGE_RANGES = [[100E-3, 100E-3], [1, 1], [10, 10], [100, 100], [1000, 1000],
+                      ["MIN", 100E-3], ["MAX", 1000], ["DEF", 10]]
+    VOLTAGE_AC_RANGES = [[100E-3, 100E-3], [1, 1], [10, 10], [100, 100], [750, 750],
+                         ["MIN", 100E-3], ["MAX", 750], ["DEF", 10]]
+    RESISTANCE_RANGES = [[1E2, 1E2], [1E3, 1E3], [1E4, 1E4], [1E5, 1E5], [1E6, 1E6],
+                         [1E7, 1E7], [1E8, 1E8], ["MIN", 1E2], ["MAX", 1E8], ["DEF", 1E3]]
+    RESISTANCE_4W_RANGES = [[1E2, 1E2], [1E3, 1E3], [1E4, 1E4], [1E5, 1E5], [1E6, 1E6],
+                            [1E7, 1E7], [1E8, 1E8], ["MIN", 1E2], ["MAX", 1E8], ["DEF", 1E3]]
+    FREQUENCY_APERTURES = [[100E-3, 100E-3], [1, 1], ["MIN", 100E-3], ["MAX", 1], ["DEF", 1]]
+    CAPACITANCE_RANGES = [[1E-9, 1E-9], [1E-8, 1E-8], [1E-7, 1E-7], [1E-6, 1E-6], [1E-5, 1E-5],
+                          [1E-4, 1E-4], [1E-3, 1E-3], [1E-2, 1E-2], ["MIN", 1E-9], ["MAX", 1E-2],
+                          ["DEF", 1E-6]]
+
+    #########
+    # TESTS #
+    #########
+
+    def test_dmm_initialization_bad(self):
+        bad_resource = "USB0::10893::45848::MY12345678::0::INSTR"
+        with pytest.raises(VisaIOError):
+            dmm = Agilent34450A(bad_resource)
+
+    def test_good_address(self):
+        dmm = Agilent34450A(self.resource)
+
+    def test_reset(self):
+        dmm = Agilent34450A(self.resource)
+        dmm.reset()
```

Comment:
Is there some query that you can use to assert the successful reset here? A boot cycle counter? Prior to the reset setting a quantity that gets reset by the device `reset()`, then checking if that happened?